File: epi_judge_cpp/snake_string.cc
#include <string>

using std::string;

string SnakeString(const string& s) {
  // Implement this placeholder.
  return "";
}

#include "test_framework/generic_test.h"

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"s"};
  GenericTestMain(args, "snake_string.tsv", &SnakeString, DefaultComparator{},
                  param_names);
  return 0;
}

File: epi_judge_java/epi/SnakeString.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class SnakeString {
  @EpiTest(testfile = "snake_string.tsv")

  public static String snakeString(String s) {
    // Implement this placeholder.
    return "";
  }

  public static void main(String[] args) {
    // The timeout is set to 15 seconds for each test case.
    // If your program ends with TIMEOUT error, and you want to try longer time
    // limit, you can extend the limit by changing the following line.
    long timeoutSeconds = 15;

    System.exit(
        GenericTest
            .runFromAnnotations(args, timeoutSeconds,
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}
